src/words.jl

```diff
@@ -104,3 +104,32 @@ function longestcommonprefix(u::Union{Word, SubWord}, v::Union{Word, SubWord})
     end
     return n
 end
+
+"""
+    issubword(u::Union{Word, SubWord}, v::Union{Word, SubWord})
```

Comment:
how does this compare to
```julia
using KnuthBendix
using BenchmarkTools

import KnuthBendix: AbstractWord

function Base.findfirst(subword::AbstractWord, word::AbstractWord)
    k = length(subword)
    for i in 1:length(word)-k+1
        @view(word[i:i+k-1]) == subword && return i
    end
    return nothing
end

Base.occursin(subword::AbstractWord, word::AbstractWord) = findfirst(subword, word) !== nothing

let W = KnuthBendix.Word(1:100), s = KnuthBendix.Word([3,2,1]) #worst case scenario
    @btime occursin($s, $W) 
end
```
could you btime?
in case of single letters (words of length 1) we could add simpler loops;

(the above allocates views on julia-1.4, but these allocations go away with julia-1.5; I'm fine with this

Reply:
That's something of a kind "was looking for cooper and found gold", here are the results: 
- **occursin**
```
let W = KnuthBendix.Word(1:100), s = KnuthBendix.Word([3,2,1]) #worst case scenario
    @btime occursin($s, $W)
end
```
1.130 μs (98 allocations: 4.59 KiB)
false

- **issubword**
```
let W = KnuthBendix.Word(1:100), s = KnuthBendix.Word([3,2,1]) #worst case scenario
    @btime KnuthBendix.issubword($s, $W)
end
```
72.532 ns (0 allocations: 0 bytes)
false

Reply:
On my computer this:
```julia
function Base.findnext(subword::AbstractWord, word::AbstractWord, pos::Integer)
    k = length(subword)
    f = first(subword)
    @inbounds for i in pos:length(word)-k+1
        # @view(word[i:i+k-1]) == subword && return i
        word[i] == f || continue
        issub = true
        for j in 2:k
            if word[i+j-1] != subword[j]
                issub = false
                break
            end
        end
        issub == true && return i:i+k-1
    end
    return nothing
end

@inline Base.findfirst(subword::AbstractWord, word::AbstractWord) = findnext(subword, word, firstindex(word))
@inline Base.occursin(subword::AbstractWord, word::AbstractWord) = findfirst(subword, word) !== nothing
```
is a constant factor slower that `issubword` (95ns vs 88ns on my computer). This is actually something we want to have reasonable performance

Reply:
Changing from `issubword` to `occursin` and purging `issubword` due to copyright discussion.